File: backend/src/models/comment.model.js
const mongoose = require('mongoose');

const CommentSchema = new mongoose.Schema({
    postId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        index: true
    },
    userId: {
        type: String,
        required: true,
        index: true
    },
    userName: {
        type: String,
        required: true,
        trim: true
    },
    userProfileImageUrl: {
        type: String
    },
    content: {
        type: String,
        required: true,
        maxlength: 2000,
        trim: true
    }
}, {
    timestamps: true
});

CommentSchema.index({ postId: 1, createdAt: -1 });

CommentSchema.set('toJSON', {
    transform: function (doc, ret) {
        ret.id = ret._id;
        delete ret._id;
        delete ret.__v;
        return ret;
    }
});

module.exports = mongoose.model('Comment', CommentSchema);

File: lib/features/home/model/comment_model.dart
/// Model representing a comment on a post
class CommentModel {
  final String? id;
  final String postId;
  final String userId;
  final String userName;
  final String? userProfileImageUrl;
  final String content;
  final DateTime createdAt;

  CommentModel({
    this.id,
    required this.postId,
    required this.userId,
    required this.userName,
    this.userProfileImageUrl,
    required this.content,
    required this.createdAt,
  });

  factory CommentModel.fromJson(Map<String, dynamic> json) {
    return CommentModel(
      id: json['id'] as String? ?? json['_id']?.toString(),
      postId: json['postId']?.toString() ?? '',
      userId: json['userId'] as String? ?? '',
      userName: json['userName'] as String? ?? 'User',
      userProfileImageUrl: json['userProfileImageUrl'] as String?,
      content: json['content'] as String? ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'] as String)
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'postId': postId,
      'userId': userId,
      'userName': userName,
      'userProfileImageUrl': userProfileImageUrl,
      'content': content,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
